### PR TITLE
{Release} Upgrade to Azure CLI 2.42.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

File: src/azure-cli-core/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+2.42.0
+++++++
+* `aaz`: Support Featur FreeFormDictType and FreeFormDictArg (#24183)
```
Comment (kairu-ms) [inline src/azure-cli-core/HISTORY.rst:8]:
```suggestion
* `aaz`: Support `FreeFormDictType` and `FreeFormDictArg` (#24183)
```

File: src/azure-cli-core/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,11 @@
...
+* `aaz`: Support configured_default in arguments (#24411)
```
Comment (kairu-ms) [inline src/azure-cli-core/HISTORY.rst:9]:
```suggestion
* `aaz`: Support `configured_default` property in arguments (#24411)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,105 @@
...
+
+**App Config**
+
+* `az appconfig`: Update raised errors in app config command module (#24400)
+
+**App Service**
+
+* Fix #23050: `az functionapp deployment source config-zip`: Fix the bug that zip deployment will fail if app settings contain any values of null (#24077)
+
+**az hdinsight**
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:34]:
```suggestion
**HDInsight**
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,105 @@
...
+
+* [BREAKING CHANGE] `az hdinsight create`: Remove the enum value 1.0 and 1.1 from the --minimal-tls-version, HDInsight doesn't support TLS version which is less than 1.2 now. (#24141)
+
+**Backup**
+
+* `az backup restore restore-disks`: Update Cross Zonal Restore behaviour for ZRS vaults and primary region CRR scenarios (#24126)
+* `az backup job show`: Change subtask start/end time from minimum value to null for ongoing or yet-to-start operation. (#24207)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:41]:
```suggestion
* `az backup job show`: Change subtask start/end time from minimum value to null for ongoing or yet-to-start operation (#24207)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,105 @@
...
+
+**Cognitive Services**
+
+* Update SDK to use latest API version (#24406)
+
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:46]: This is not customer-facing change. 

```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,105 @@
...
+* `az sig image-version update`: Support `excludeFromLatest` for `--add` parameter to exclude this image version when using the latest version of image definition (#24412)
+* `az sig image-version update`: Support `safetyProfile.allowDeletionOfReplicatedLocations` for `--set` parameter to allow users to remove the gallery image version from replicated regions (#24412)
+
+**IoT**
+
+* `az iot hub create`: Enforce data residency property on hubs created in `qatarcentral` (#24212)
+
+**NetAppFiles**
+
+* `az netappfiles account renew-credentials`: Add `renew-credentials` comand to renew identity credentials that are used to authenticate to key vault, for customer-managed key encryption (#24423)
```
Comment (bebound) [inline src/azure-cli/HISTORY.rst:63]:
```suggestion
* `az netappfiles account renew-credentials`: Add `renew-credentials` command to renew identity credentials that are used to authenticate to key vault, for customer-managed key encryption (#24423)
```
`that are used to authenticate to key vault, for customer-managed key encryption`  can also be removed?

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,105 @@
...
+
+* [BREAKING CHANGE] `az postgres flexible-server migration show`: Remove `--level` parameter (#24055)
+* [BREAKING CHANGE] `az postgres flexible-server migration delete`: Remove this command. Deleting a migration is not supported for now. (#24055)
+* [BREAKING CHANGE] Change `az postgres flexible-server migration update --cutover` to `az postgres flexible-server migration update --cutover db1 db2 db3` (#24055)
+* `az postgres flexible-server migration create`: Add `--migration-mode` to support offline and online(with CDC) migrations. Default mode when `--migration-mode` not passed will be offline. (#24055)
+* Add `az postgres flexible-server migration update --cancel db1 db2 db3` to cancel a migration. (#24055)
+
+**Resource**
+
+* `az resource delete`: Add new parameters `--no-wait` to support not waiting the long-running operation to finish (#24302)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:84]:
```suggestion
* `az resource delete`: Add new parameter `--no-wait` to support not waiting the long-running operation to finish (#24302)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,105 @@
 Release History
 ===============
 
+2.42.0
+++++++
+
+**ACR**
+
+* `acr task update`: Fix logic issue for updating encoded task (#24279)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:11]:
```suggestion
* `az acr task update`: Fix logic issue for updating encoded task (#24279)
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,105 @@
...
+
+* `az appconfig`: Update raised errors in app config command module (#24400)
+
+**App Service**
+
+* Fix #23050: `az functionapp deployment source config-zip`: Fix the bug that zip deployment will fail if app settings contain any values of null (#24077)
+
+**az hdinsight**
+
+* [BREAKING CHANGE] `az hdinsight create`: Remove the enum value 1.0 and 1.1 from the --minimal-tls-version, HDInsight doesn't support TLS version which is less than 1.2 now. (#24141)
```
Comment (wangzelin007) [inline src/azure-cli/HISTORY.rst:36]:
```suggestion
* [BREAKING CHANGE] `az hdinsight create`: Remove the enum value 1.0 and 1.1 from the `--minimal-tls-version`, HDInsight doesn't support TLS version which is less than 1.2 now. (#24141)
```

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,105 @@
...
+* Fix #24188: `az aks list`: Fix pagination handling error `ContainerServiceClientConfiguration object has no attribute api_version` when there are many list results (#24270)
+* Fix #24188: `az aks nodepool list`: Fix pagination handling error `ContainerServiceClientConfiguration object has no attribute api_version` when there are many list results (#24270)
+* `az aks create/update`: Add new parameters `--enable-blob-driver` and `--disable-blob-driver` to enable/disable Blob CSI Driver (#24404)
+* `az aks create/update`: Add new parameter `--enable-oidc-issuer` to support enabling oidc issuer feature (#24070)
+* `az aks oidc-issuer rotate-signing-keys`: Add new command to support rotating oidc issuer service account signing keys (#24070)
+
+**APIM**
+
+* `apim create/update`: Add `--public-network-access` to support specifying whether or not public endpoint access is allowed for this API management service (#23983)
+* `apim create/update`: Add `--disable-gateway` to support disabling gateway in the master region (#23983)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:24]:
```suggestion
* `az apim create/update`: Add `--public-network-access` to support specifying whether or not public endpoint access is allowed for this API management service (#23983)
* `az apim create/update`: Add `--disable-gateway` to support disabling gateway in the master region (#23983)
```

Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:24]:
```suggestion
* `az apim create/update`: Add `--public-network-access` to support specifying whether or not public endpoint access is allowed for this API management service (#23983)
* `az apim create/update`: Add `--disable-gateway` to support disabling gateway in the master region (#23983)
```

File: src/azure-cli/HISTORY.rst (hunk 8 above)
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:11]:
```suggestion
* `az acr task update`: Fix logic issue for updating encoded task (#24279)
```

**Review by wangzelin007 — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Comment by yonzhan** [comment]
Release

**Pull request merged**